### PR TITLE
Support agent ID parameter

### PR DESCRIPTION
**Pull request by code-asher**
Part of https://github.com/coder/jetbrains-coder/issues/292

This will let you set the agent by either ID or name.  If both are set then the name is ignored.

### CODE REVIEW

File: src/main/kotlin/com/coder/gateway/CoderGatewayConnectionProvider.kt [hunk 1]
```diff
@@ -72,14 +73,23 @@ class CoderGatewayConnectionProvider : GatewayConnectionProvider {
...
-                // TODO: Show a dropdown and ask for an agent.
-                throw IllegalArgumentException("Query parameter \"$AGENT\" is missing")
+                if (parameters[AGENT_ID].isNullOrBlank() && parameters[AGENT_NAME].isNullOrBlank()) {
+                    // TODO: Show a dropdown and ask for an agent.
+                    throw IllegalArgumentException("Unable to determine which agent to connect to; one of \"$AGENT_NAME\" or \"$AGENT_ID\" must be set because \"$workspaceName\" has more than one agent")
+                } else if (parameters[AGENT_ID].isNullOrBlank()) {
+                    throw IllegalArgumentException("The workspace \"$workspaceName\" does not have an agent with ID \"${parameters[AGENT_ID]}\"")
+                } else {
+                    throw IllegalArgumentException("The workspace \"$workspaceName\" does not have an agent named \"${parameters[AGENT_NAME]}\"")
+                }
```
Comment (johnstcn) [inline src/main/kotlin/com/coder/gateway/CoderGatewayConnectionProvider.kt:92]: This is starting to look like it wants to be extracted to its own function.

File: src/test/groovy/DataGen.groovy [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
 
 class DataGen {
-    static WorkspaceAgentModel workspace(String name, String workspaceName = name) {
+    // Create a random workspace agent model.  If the workspace name is omitted
+    // then return a model without any agent bits, similar to what
+    // toAgentModels() does if the workspace does not specify any agents.
+    // TODO: Maybe better to randomly generate the workspace and then call
+    //       toAgentModels() on it.  Also the way an "agent" model can have no
+    //       agent in it seems weird; can we refactor to remove
+    //       WorkspaceAgentModel and use the original structs from the API?
```
Comment (johnstcn) [inline src/test/groovy/DataGen.groovy:13]: 👍 to this, I'm assuming we auto-generate the API? I wonder if we could gen the field values from the spec.

**Review by johnstcn — APPROVED** [review]

File: src/test/groovy/DataGen.groovy (hunk 2 above)
Comment (code-asher) [reply src/test/groovy/DataGen.groovy:13]: Oh that would be absolutely wonderful.  The API types appear to be hand-written at the moment.

Comment (code-asher) [reply src/test/groovy/DataGen.groovy:13]: https://github.com/coder/jetbrains-coder/issues/314

**Pull request merged**